cmd/argocd-applicationset-controller/commands/applicationset_controller.go

```diff
@@ -208,7 +213,18 @@ func NewCommand() *cobra.Command {
 				func(appset *appv1alpha1.ApplicationSet) bool {
 					return utils.IsNamespaceAllowed(applicationSetNamespaces, appset.Namespace)
 				})
-
+			cache, err := cacheSource()
+			errors.CheckError(err)
+			var appStatusProvider status.AppResourceStatusProvider
+			if persistResourceHealth {
```

Comment:
Instead of passing down `persistResourceHealth` I could also check each application for `ResourceHealthLocationAppTree` and if one application has it, grab all the resources from redis.